shoop/utils/properties.py

```diff
@@ -0,0 +1,178 @@
+
+    def __repr__(self):
+        argstr = ', '.join('%s=%r' % x for x in self._fields.items())
+        return "%s(%s)" % (type(self).__name__, argstr)
+
+    def __get__(self, instance, type=None):
+        if instance is None:
+            return self
+        return self._get_value_from(instance)
+
+    def _get_value_from(self, instance, overrides={}):
+        data = {
+            field: resolve(instance, path)
+            for (field, path) in self._fields.items()
+        }
+        data.update(overrides)
+        if data['value'] is None:
+            return None
+        return self.value_class.from_data(**data)
+
+    def __set__(self, instance, value):
+        if value is not None:
+            self._check_unit(instance, value)
+        self._set_part(instance, 'value', value)
+
+    def _check_unit(self, instance, value):
+        value_template = self._get_value_from(instance, overrides={'value': 0})
+        if not value_template.unit_matches_with(value):
+            msg = 'Cannot set %s to value with non-matching unit' % (
+                type(self).__name__,)
+            raise UnitMixupError(value_template, value, msg)
+        assert isinstance(value, self.value_class)
+
+    def _set_part(self, instance, part_name, value):
+        value_full_path = self._fields[part_name]
+        if '.' in value_full_path:
+            (obj_path, attr_to_set) = value_full_path.rsplit('.', 1)
+            obj = resolve(instance, obj_path)
+        else:
+            attr_to_set = value_full_path
+            obj = instance
+        if value is not None:
+            setattr(obj, attr_to_set, getattr(value, part_name))
+        else:
+            setattr(obj, attr_to_set, None)
+
+
+class PriceProperty(MoneyProperty):
+    """
+    Property for Price object.
+
+    Similar to `MoneyProperty` but also has ``includes_tax`` field.
+
+    Operaters with `TaxfulPrice` and `TaxlessPrice` objects.
+    """
+    value_class = Price
+
+    def __init__(self, value, currency, includes_tax, **kwargs):
+        """
+        Initialize PriceProperty with given field locators.
+
+        :param value: Locator for value of the Price
+        :type value: str
+        :param currency: Locator for currency of the Price
+        :type currency: str
+        :param includes_tax: Locator for includes_tax of the Price
+        :type includes_tax: str
+        """
+        super(PriceProperty, self).__init__(value, currency, **kwargs)
+        self._fields['includes_tax'] = includes_tax
+
+
+class TaxfulPriceProperty(MoneyProperty):
+    value_class = TaxfulPrice
+
+
+class TaxlessPriceProperty(MoneyProperty):
+    value_class = TaxlessPrice
+
+
+class MoneyPropped(object):
+    """
+    Mixin for transforming MoneyProperty init parameters.
+
+    Add this mixin as (first) base for the class that has
+    `MoneyProperty` properties and this will make its `__init__`
+    tranform passed kwargs to the fields specified in the
```

Comment:
Seemz that I have a problem with that letter, maybe it'z broken on my keyboard..